File: backend/internal/server/dto/oneclick.go
package dto

// OneClickIssueTokenReq 签发一次性接入令牌请求。
type OneClickIssueTokenReq struct {
	KeyID int64 `json:"key_id" binding:"required,min=1"`
}

// OneClickIssueTokenResp 签发结果：带按平台拼好的完整接入命令，前端直接展示复制。
type OneClickIssueTokenResp struct {
	Token             string `json:"token"`
	ExpiresInSeconds  int    `json:"expires_in_seconds"`
	BaseURL           string `json:"base_url"`
	CommandBash       string `json:"command_bash"`
	CommandPowerShell string `json:"command_powershell"`
}

// OneClickStatusResp 令牌状态轮询响应。status: pending | exchanged | verified | expired。
type OneClickStatusResp struct {
	Status string `json:"status"`
}

File: web/src/shared/api/oneclick.ts
import { get, post } from './client';

// 一键接入（Claude Code）：签发一次性 setup token + 轮询接入状态。
// 后端对应 /api/v1/oneclick/*（JWT 鉴权）；脚本侧的 /oneclick/* 公开端点前端不直接调用。

export interface OneClickIssueResp {
  token: string;
  expires_in_seconds: number;
  base_url: string;
  command_bash: string;
  command_powershell: string;
  command_codex_bash: string;
  command_codex_powershell: string;
}

export type OneClickStatus = 'pending' | 'exchanged' | 'verified' | 'expired';

export interface OneClickStatusResp {
  status: OneClickStatus;
}

export const oneclickApi = {
  issue: (keyId: number) =>
    post<OneClickIssueResp>('/api/v1/oneclick/setup-token', { key_id: keyId }),
  status: (token: string) =>
    get<OneClickStatusResp>(`/api/v1/oneclick/setup-token/${token}`),
};
